sda-commons-server-circuitbreaker/build.gradle

```diff
@@ -7,6 +7,8 @@ dependencies {
     // version clash with latest client 0.9.0 -> 0.16.0
     exclude group: 'io.prometheus', module: 'simpleclient_common'
   }
+  implementation 'io.micrometer:micrometer-core'
```

Comment:
We haven't put much effort into the distinction between "api" and "implementation". Will users of dw-commons be able to track metrics? Or is that dependency necessary?

Reply:
My understanding is that using the `implementation` configuration users of dw-commons will not be able to use micrometer-core code unless they add the library explicitly.

discussed this a couple of months ago with @JoergSiebahn and decided then that we don't want to expose micrometer to users of our library but just use it internally till further discussions happened.
Since we decided to move forward with micrometer we could now also expose the library to make it available to users but should then do it consistently 
https://github.com/SDA-SE/sda-dropwizard-commons/blob/c301f7806f9fbf72444271cba9f6b8169f751e1b/sda-commons-server-kafka/build.gradle#L9



@JoergSiebahn any objections using `api` here and make micrometer available to users of our library?

Reply:
I verified that we still see the implementation in the generated pom, but with `runtime` scope:
```
<dependency>
      <groupId>io.micrometer</groupId>
      <artifactId>micrometer-core</artifactId>
      <scope>runtime</scope>
      …
```
So, using `implementation` here would be ok for me. 😄

Reply:
Imho we only should use API where consumers need to interact with the 3rd party code. As this was not handled very consistently in the past, I would not change everything but start going in the right direction when possible. As we already use implementation here, I would keep it like that. Afaik you won't use micrometer for custom service specific metrics but stick with the Dropwizard metrics in that case?

Reply:
fair enough then we leave it as is